crates/biome_js_analyze/src/analyzers/nursery/no_duplicate_test_hooks.rs

```diff
@@ -0,0 +1,221 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, AddVisitor, Phases, QueryMatch, Queryable, Rule,
+    RuleDiagnostic, ServiceBag, Visitor, VisitorContext,
+};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsCallExpression, JsLanguage, JsSyntaxToken, TextRange};
+use biome_rowan::{AstNode, Language, SyntaxNode, WalkEvent};
+
+declare_rule! {
+    /// A `describe` block should not contain duplicate hooks.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// describe('foo', () => {
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   test('foo_test', () => {
+    ///    // some test
+    ///   });
+    /// });
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// describe('foo', () => {
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   test('foo_test', () => {
+    ///     afterAll(() => {
+    ///       // some teardown
+    ///     });
+    ///    afterAll(() => {
+    ///      // some teardown
+    ///    });
+    ///   });
+    /// });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// describe('foo', () => {
+    ///   beforeEach(() => {
+    ///     // some setup
+    ///   });
+    ///   test('foo_test', () => {
+    ///     // some test
+    ///   });
+    /// });
+    /// ```
+    ///
+    pub(crate) NoDuplicateTestHooks {
+        version: "next",
+        name: "noDuplicateTestHooks",
+        recommended: true,
+    }
+}
+
+#[derive(Debug, Default)]
+struct HooksContext {
+    before_all: usize,
+    before_each: usize,
+    after_all: usize,
+    after_each: usize,
+}
```

Comment:
This looks good, although this rule should also check for `before` and `after` too. We don't want to make the rule a perfect match with the Jest eslint rule, because we want to cover other testing frameworks too.